Geometry/Test/FindIntersections.test.js

```diff
@@ -0,0 +1,159 @@
+import findIntersections, {
+  createPoint,
+  createEvent,
+  comparePoints,
+  compareSegmentsY,
+  calculateIntersectionPoint,
+  handlePotentialIntersection
+} from '../FindIntersections'
+
+describe('Geometry Functions', () => {
+  describe('createPoint', () => {
+    it('should create a point with given coordinates', () => {
+      const point = createPoint(1, 2)
+      expect(point).toEqual({ x: 1, y: 2 })
+    })
+  })
+
+  describe('createEvent', () => {
+    it('should create an event with the given parameters', () => {
+      const point = createPoint(1, 2)
+      const event = createEvent(point, ['left'], [])
+      expect(event).toEqual({
+        point,
+        types: ['left'],
+        segments: []
+      })
+    })
+  })
+
+  describe('comparePoints', () => {
+    it('should return negative if first point is less than second', () => {
+      const a = createPoint(1, 2)
+      const b = createPoint(2, 3)
+      expect(comparePoints(a, b)).toBeLessThan(0)
+    })
+
+    it('should return positive if first point is greater than second', () => {
+      const a = createPoint(2, 3)
+      const b = createPoint(1, 2)
+      expect(comparePoints(a, b)).toBeGreaterThan(0)
+    })
+
+    it('should return zero if points are equal', () => {
+      const a = createPoint(1, 2)
+      const b = createPoint(1, 2)
+      expect(comparePoints(a, b)).toBe(0)
+    })
+  })
+
+  describe('compareSegmentsY', () => {
+    it('should compare segments based on their start y-coordinates', () => {
+      const seg1 = { start: { x: 0, y: 1 }, end: { x: 1, y: 1 } }
+      const seg2 = { start: { x: 0, y: 2 }, end: { x: 1, y: 2 } }
+      expect(compareSegmentsY(seg1, seg2)).toBe(-1)
+    })
+
+    it('should return zero if segments have the same start y-coordinate', () => {
+      const seg1 = { start: { x: 0, y: 1 }, end: { x: 1, y: 1 } }
+      const seg2 = { start: { x: -1, y: 1 }, end: { x: -2, y: 1 } }
+      expect(compareSegmentsY(seg1, seg2)).toBe(0)
+    })
+
+    it('should return positive if first segment is greater than second', () => {
+      const seg1 = { start: { x: 0, y: 3 }, end: { x: 1, y: 3 } }
+      const seg2 = { start: { x: -1, y: 2 }, end: { x: -2, y: 2 } }
+      expect(compareSegmentsY(seg1, seg2)).toBe(1)
+    })
+  })
+
+  describe('calculateIntersectionPoint', () => {
```

Comment:
Only `calculateIntersectionPoint`, `handlePotentialIntersection` and `findIntersections` really neeed tests. The rest is practically trivial and also covered transitively by sufficiently complex tests of the latter.